### PR TITLE
a blog post for the docker machine stuff

### PR DESCRIPTION
**Pull request by simonjj**
Simon J is a product manager at Rackspace. He's responsible for Rackspace's container offerings and strategy.

Edits are perfectly fine.

Hopefully this will go public on Monday.

### CODE REVIEW

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+comments: true
+author: Simon J
+published: true
+categories:
+    - Orchestration
+    - docker
+---
+
+## Background
+Beginning of December Docker anounnced [Docker Machine](http://blog.docker.com/2014/12/announcing-docker-machine-swarm-and-compose-for-orchestrating-distributed-apps/). Docker Machine does away with the need to first create virtual machines, install Docker on them, to finally use them as a Docker host. Instead it covers the provisioning and install process for you behind the scences. To read more about Docker Machines [click here](https://github.com/docker/machine).
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: At the beginning of December, Docker ...
... Docker Machine does away with the need to create virtual machines, then to install Docker on them, and finally to use them as a Docker host. Instead, covers the provisioning and installation process for you, behind the ...

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+### Install
+
+Docker Machine is currently still unreleased, which means you have to first download the machine binary from [https://github.com/docker/machine/releases](https://github.com/docker/machine/releases). There aren't any dependencies to install, so you can just rename it to docker-machine and place it into your path somewhere.
+
+
+### Credentials
+
+Unsurprisingly you'll also need your API-key and Rackspace username. If you're unsure about how or where to get those please follow [these instructions](http://www.rackspace.com/knowledge_center/article/view-and-reset-your-api-key).
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: Unsurprisingly, you'll also need your API-key and Rackspace username. If you're unsure about how or where to get those, please follow ...

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+We can now user docker-machine to create our Docker host:
+
+
+      docker-machine create --driver rackspace \
+                            --rackspace-api-key $YOUR_API_KEY \
+                            --rackspace-username $YOUR_USERNAME \
+                            --rackspace-region DFW my-first-docker-machine-host
+
+This step will take a few minutes. If you want to streamline your command further you can export your credentials and region via `OS_USERNAME`, `OS_API_KEY` and `OS_REGION_NAME`. This reduces the command to:
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: ... If you want to streamline your command further, you can export ...

**Comment by rdodev** [comment]
@simonjj @catlook is our blog copy editor. She'll suggest grammar and wording changes and then up to you whether you want to incorporate them or not.

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+
+    docker-machine create --driver rackspace my-second-docker-machine-host
+
+In either case you should see something along the lines of:
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: In either case, you should ...

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    # removing a machine
+    docker-machine rm my-first-docker-machine-host
+
+More information can be [found here](https://github.com/docker/machine#rackspace).
+
+
+### Lets actually put this to work
+
+Lastly we want to actually use our machine to deploy a container. To do so we simply use the `docker` command, we extract the matching config via the docker-machine command:
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: Lastly, we want to ...
... To do so ,we simply use the `docker` command and extract the ...

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    docker `docker-machine config my-second-docker-machine-host` run --name my-redis-container -d redis
+
+We can then run the `ps` command to check on our container:
+
+    docker `docker-machine config my-second-docker-machine-host` ps
+
+    CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS               NAMES
+    3cff35053aee        redis:latest        "/entrypoint.sh redi   35 minutes ago      Up 5 minutes       6379/tcp            my-redis-container
+
+and finally we confirm that Redis is indeed running and accepting connections. In order to do so we spin up a mini container, drop to the console, and use telnet to see if Redis `PINGs`:
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: Finally, we confirm that ...
... In order to do so, we spin up ...

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    docker `docker-machine config my-second-docker-machine-host` run --link my-redis-container:redis -it --rm busybox
+
+    / >>> telnet redis 6379
+    PING
+    +PONG
+
+
+If you have feedback please contact us via [sdk-support@rackspace.com](mailto:sdk-support@rackspace.com), twitter [@rackspace](https://twitter.com/rackspace), chat with us in #rackspace on Freenode, or post in our [developer forum](https://community.rackspace.com/developers/default).
```
Comment (catlook) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: If you have feedback, please contact us ...

[Nice article!!! Thanks for sharing.  Feel free free to use any of my edits that you like and discard the rest :)]

File: src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title: "Automate deploying Rackspace Cloud Monitoring agent"
```
Comment (rgbkrk) [inline src/site_source/_posts/2015-01-26-using-docker-machine-to-deploy-your-docker-containers-on-rackspace.md]: Delete this extra title out.

**Comment by adrianotto** [comment]
-1

Consider edited version here:
https://github.com/rackerlabs/developer.rackspace.com/pull/1033

**Comment by rdodev** [comment]
As per @adrianotto I'm closing this PR in favor of the he pointed out.

**Pull request closed without merging**